<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>mmtbx.building package &mdash; CCTBX None documentation</title>
    
    <link rel="stylesheet" href="../_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     'None',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="CCTBX None documentation" href="../index.html" />
    <link rel="up" title="mmtbx - macromolecular crystallography" href="mmtbx.html" />
    <link rel="next" title="mmtbx.building.alternate_conformations package" href="mmtbx.building.alternate_conformations.html" />
    <link rel="prev" title="mmtbx - macromolecular crystallography" href="mmtbx.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="../py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="mmtbx.building.alternate_conformations.html" title="mmtbx.building.alternate_conformations package"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="mmtbx.html" title="mmtbx - macromolecular crystallography"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">CCTBX None documentation</a> &raquo;</li>
          <li class="nav-item nav-item-1"><a href="mmtbx.html" accesskey="U">mmtbx - macromolecular crystallography</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="mmtbx-building-package">
<h1>mmtbx.building package<a class="headerlink" href="#mmtbx-building-package" title="Permalink to this headline">¶</a></h1>
<div class="section" id="subpackages">
<h2>Subpackages<a class="headerlink" href="#subpackages" title="Permalink to this headline">¶</a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="mmtbx.building.alternate_conformations.html">mmtbx.building.alternate_conformations package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="mmtbx.building.alternate_conformations.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="mmtbx.building.alternate_conformations.html#module-mmtbx.building.alternate_conformations.altloc_labeling">mmtbx.building.alternate_conformations.altloc_labeling module</a><ul class="simple">
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="mmtbx.building.alternate_conformations.html#module-mmtbx.building.alternate_conformations.single_residue">mmtbx.building.alternate_conformations.single_residue module</a><ul class="simple">
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="mmtbx.building.alternate_conformations.html#module-mmtbx.building.alternate_conformations">Module contents</a><ul class="simple">
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section" id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="module-mmtbx.building.extend_sidechains">
<span id="mmtbx-building-extend-sidechains-module"></span><h2>mmtbx.building.extend_sidechains module<a class="headerlink" href="#module-mmtbx.building.extend_sidechains" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="mmtbx.building.extend_sidechains.conformation_scorer">
<em class="property">class </em><code class="descclassname">mmtbx.building.extend_sidechains.</code><code class="descname">conformation_scorer</code><span class="sig-paren">(</span><em>old_residue</em>, <em>new_residue</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.conformation_scorer" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/2/library/functions.html#object" title="(in Python v2.7)"><code class="xref py py-class docutils literal"><span class="pre">object</span></code></a></p>
<p>Stand-in for the conformation scoring class in mmtbx.refinement.real_space;
instead of calculating fit to the map, this simply uses the change in
position of the first atom being moved at each rotation.  This allows us to
superimpose the conformations for those atoms which are present in both the
old and the new residues.</p>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.extend_sidechains.conformation_scorer.apply_final" title="mmtbx.building.extend_sidechains.conformation_scorer.apply_final"><code class="xref py py-obj docutils literal"><span class="pre">apply_final</span></code></a>()</td>
<td></td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mmtbx.building.extend_sidechains.conformation_scorer.reset" title="mmtbx.building.extend_sidechains.conformation_scorer.reset"><code class="xref py py-obj docutils literal"><span class="pre">reset</span></code></a>(sites_cart,&nbsp;selection)</td>
<td></td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.extend_sidechains.conformation_scorer.update" title="mmtbx.building.extend_sidechains.conformation_scorer.update"><code class="xref py py-obj docutils literal"><span class="pre">update</span></code></a>(sites_cart,&nbsp;selection)</td>
<td></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="mmtbx.building.extend_sidechains.conformation_scorer.apply_final">
<code class="descname">apply_final</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.conformation_scorer.apply_final" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.extend_sidechains.conformation_scorer.reset">
<code class="descname">reset</code><span class="sig-paren">(</span><em>sites_cart</em>, <em>selection</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.conformation_scorer.reset" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.extend_sidechains.conformation_scorer.update">
<code class="descname">update</code><span class="sig-paren">(</span><em>sites_cart</em>, <em>selection</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.conformation_scorer.update" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="function">
<dt id="mmtbx.building.extend_sidechains.correct_sequence">
<code class="descclassname">mmtbx.building.extend_sidechains.</code><code class="descname">correct_sequence</code><span class="sig-paren">(</span><em>pdb_hierarchy</em>, <em>sequences</em>, <em>truncate_to_cbeta=False</em>, <em>out=&lt;open file '&lt;stdout&gt;'</em>, <em>mode 'w'&gt;</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.correct_sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Modify the sequence for the pdb hierarchy to match that of the aligned
sequence.  This will remove incompatible atoms; the sidechains will still
need to be extended separated.  For proteins only - mismatches in nucleic
acids will only result in a warning.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>pdb_hierarchy</strong> &#8211; iotbx.pdb.hierarchy.root object</li>
<li><strong>sequences</strong> &#8211; list of iotbx.bioinformatics.sequence objects</li>
<li><strong>trucate_to_cbeta</strong> &#8211; chop off entire sidechain to C-beta (default: leave
common atoms in place)</li>
<li><strong>out</strong> &#8211; output filehandle (default = stdout)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">number of atom_group objects renamed</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="mmtbx.building.extend_sidechains.extend_and_refine">
<em class="property">class </em><code class="descclassname">mmtbx.building.extend_sidechains.</code><code class="descname">extend_and_refine</code><span class="sig-paren">(</span><em>pdb_hierarchy</em>, <em>xray_structure</em>, <em>fmodel</em>, <em>params</em>, <em>cif_objects=()</em>, <em>out=&lt;open file '&lt;stdout&gt;'</em>, <em>mode 'w'&gt;</em>, <em>output_model=None</em>, <em>output_map_coeffs=None</em>, <em>prefix=None</em>, <em>write_files=True</em>, <em>reset_segid=True</em>, <em>verbose=True</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.extend_and_refine" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/2/library/functions.html#object" title="(in Python v2.7)"><code class="xref py py-class docutils literal"><span class="pre">object</span></code></a></p>
<p>Run the combined sidechain extension and real-space fitting, and optionally
write final model and map coefficients.</p>
</dd></dl>

<dl class="function">
<dt id="mmtbx.building.extend_sidechains.extend_protein_model">
<code class="descclassname">mmtbx.building.extend_sidechains.</code><code class="descname">extend_protein_model</code><span class="sig-paren">(</span><em>pdb_hierarchy</em>, <em>mon_lib_srv</em>, <em>add_hydrogens=None</em>, <em>selection=None</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.extend_protein_model" title="Permalink to this definition">¶</a></dt>
<dd><p>Rebuild a sidechain by substituting an ideal amino acid and rotating the
sidechain to match the old conformation as closely as possible.
Limited functionality:</p>
<blockquote>
<div><ol class="arabic simple">
<li>Amino-acids only, 2) side chain atoms only.</li>
</ol>
</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="mmtbx.building.extend_sidechains.extend_residue">
<code class="descclassname">mmtbx.building.extend_sidechains.</code><code class="descname">extend_residue</code><span class="sig-paren">(</span><em>residue</em>, <em>target_atom_group</em>, <em>mon_lib_srv</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.extend_residue" title="Permalink to this definition">¶</a></dt>
<dd><p>Rebuild a sidechain by substituting an ideal amino acid and rotating the
sidechain to match the old conformation as closely as possible.
Limited functionality:</p>
<blockquote>
<div><ol class="arabic simple">
<li>Amino-acids only, 2) side chain atoms only.</li>
</ol>
</div></blockquote>
</dd></dl>

<dl class="class">
<dt id="mmtbx.building.extend_sidechains.prefilter">
<em class="property">class </em><code class="descclassname">mmtbx.building.extend_sidechains.</code><code class="descname">prefilter</code><span class="sig-paren">(</span><em>fmodel</em>, <em>out</em>, <em>backbone_min_sigma=1.0</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.prefilter" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/2/library/functions.html#object" title="(in Python v2.7)"><code class="xref py py-class docutils literal"><span class="pre">object</span></code></a></p>
<p>Optional filter for excluding residues with poor backbone density from being
extended.  This is done as a separate callback to enable the main rebuilding
routine to be independent of data/maps.</p>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><code class="xref py py-obj docutils literal"><span class="pre">__call__</span></code>(residue)</td>
<td></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="mmtbx.building.extend_sidechains.refit_residues">
<code class="descclassname">mmtbx.building.extend_sidechains.</code><code class="descname">refit_residues</code><span class="sig-paren">(</span><em>pdb_hierarchy</em>, <em>cif_objects</em>, <em>fmodel</em>, <em>use_rotamers=True</em>, <em>anneal=False</em>, <em>verbose=True</em>, <em>allow_modified_residues=False</em>, <em>out=&lt;open file '&lt;stdout&gt;'</em>, <em>mode 'w'&gt;</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.refit_residues" title="Permalink to this definition">¶</a></dt>
<dd><p>Use real-space refinement tools to fit newly extended residues.</p>
</dd></dl>

</div>
<div class="section" id="module-mmtbx.building.make_library">
<span id="mmtbx-building-make-library-module"></span><h2>mmtbx.building.make_library module<a class="headerlink" href="#module-mmtbx.building.make_library" title="Permalink to this headline">¶</a></h2>
<p>Tools for assembling an ensemble of related structures for local rebuilding by
homology.</p>
<dl class="class">
<dt id="mmtbx.building.make_library.ensembler">
<em class="property">class </em><code class="descclassname">mmtbx.building.make_library.</code><code class="descname">ensembler</code><span class="sig-paren">(</span><em>reference_hierarchy</em>, <em>related_chains</em>, <em>atom_selection_string=None</em>, <em>backbone_only=&lt;libtbx.AutoType object&gt;</em>, <em>calpha_only=None</em>, <em>nproc=&lt;libtbx.AutoType object&gt;</em>, <em>sieve_fit=False</em>, <em>frac_discard=0.5</em>, <em>log=&lt;libtbx.utils.null_out object&gt;</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.ensembler" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/2/library/functions.html#object" title="(in Python v2.7)"><code class="xref py py-class docutils literal"><span class="pre">object</span></code></a></p>
<p>Superpose a collection of single-chain models on a reference chain.  This is
intentionally very limited in function, but flexible with respect to the
choice of atoms to superpose.</p>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.make_library.ensembler.align_model" title="mmtbx.building.make_library.ensembler.align_model"><code class="xref py py-obj docutils literal"><span class="pre">align_model</span></code></a>(i_model)</td>
<td></td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mmtbx.building.make_library.ensembler.as_multi_model_hierarchy" title="mmtbx.building.make_library.ensembler.as_multi_model_hierarchy"><code class="xref py py-obj docutils literal"><span class="pre">as_multi_model_hierarchy</span></code></a>()</td>
<td></td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.make_library.ensembler.realign" title="mmtbx.building.make_library.ensembler.realign"><code class="xref py py-obj docutils literal"><span class="pre">realign</span></code></a>(atom_selection_string)</td>
<td></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="mmtbx.building.make_library.ensembler.align_model">
<code class="descname">align_model</code><span class="sig-paren">(</span><em>i_model</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.ensembler.align_model" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.make_library.ensembler.as_multi_model_hierarchy">
<code class="descname">as_multi_model_hierarchy</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.ensembler.as_multi_model_hierarchy" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.make_library.ensembler.realign">
<code class="descname">realign</code><span class="sig-paren">(</span><em>atom_selection_string</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.ensembler.realign" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="function">
<dt id="mmtbx.building.make_library.extract_and_superpose">
<code class="descclassname">mmtbx.building.make_library.</code><code class="descname">extract_and_superpose</code><span class="sig-paren">(</span><em>reference_hierarchy</em>, <em>search_directory</em>, <em>sequence</em>, <em>params</em>, <em>nproc=&lt;libtbx.AutoType object&gt;</em>, <em>out=&lt;open file '&lt;stdout&gt;'</em>, <em>mode 'w'&gt;</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.extract_and_superpose" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="mmtbx.building.make_library.extract_peptide_fragments_by_sequence">
<code class="descclassname">mmtbx.building.make_library.</code><code class="descname">extract_peptide_fragments_by_sequence</code><span class="sig-paren">(</span><em>pdb_hierarchy</em>, <em>sequence</em>, <em>renumber_from=None</em>, <em>mainchain_only=False</em>, <em>remove_hydrogens=True</em>, <em>reset_chain_id='A'</em>, <em>reset_icode=' '</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.extract_peptide_fragments_by_sequence" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="mmtbx.building.make_library.extract_related_models">
<em class="property">class </em><code class="descclassname">mmtbx.building.make_library.</code><code class="descname">extract_related_models</code><span class="sig-paren">(</span><em>sources_and_models</em>, <em>sequence</em>, <em>first_chain_only=False</em>, <em>reset_chain_id=None</em>, <em>min_identity=0.95</em>, <em>nproc=&lt;libtbx.AutoType object&gt;</em>, <em>log=&lt;libtbx.utils.null_out object&gt;</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.extract_related_models" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/2/library/functions.html#object" title="(in Python v2.7)"><code class="xref py py-class docutils literal"><span class="pre">object</span></code></a></p>
<p>Multiprocessing wrapper for pulling matching chains out of a collection of
PDB hierarchies (either from local files or from a PDB mirror).</p>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.make_library.extract_related_models.examine_model" title="mmtbx.building.make_library.extract_related_models.examine_model"><code class="xref py py-obj docutils literal"><span class="pre">examine_model</span></code></a>(i_model)</td>
<td></td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mmtbx.building.make_library.extract_related_models.results" title="mmtbx.building.make_library.extract_related_models.results"><code class="xref py py-obj docutils literal"><span class="pre">results</span></code></a>()</td>
<td></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="mmtbx.building.make_library.extract_related_models.examine_model">
<code class="descname">examine_model</code><span class="sig-paren">(</span><em>i_model</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.extract_related_models.examine_model" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.make_library.extract_related_models.results">
<code class="descname">results</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.extract_related_models.results" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="function">
<dt id="mmtbx.building.make_library.fetch_similar_pdb_ids">
<code class="descclassname">mmtbx.building.make_library.</code><code class="descname">fetch_similar_pdb_ids</code><span class="sig-paren">(</span><em>sequence</em>, <em>min_identity=0.95</em>, <em>min_resolution=3.0</em>, <em>xray_only=True</em>, <em>data_only=False</em>, <em>expect=1e-07</em>, <em>sort_by_resolution=True</em>, <em>ligands=None</em>, <em>log=None</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.fetch_similar_pdb_ids" title="Permalink to this definition">¶</a></dt>
<dd><p>Finds closely related structures in the RCSB database, optionally limited to
structures with specific ligands.</p>
</dd></dl>

<dl class="function">
<dt id="mmtbx.building.make_library.find_similar_chains">
<code class="descclassname">mmtbx.building.make_library.</code><code class="descname">find_similar_chains</code><span class="sig-paren">(</span><em>pdb_hierarchy</em>, <em>sequence</em>, <em>source_info=None</em>, <em>first_chain_only=False</em>, <em>remove_alt_confs=True</em>, <em>reset_chain_id=None</em>, <em>min_identity=0.95</em>, <em>min_identity_epsilon=0.02</em>, <em>log=&lt;libtbx.utils.null_out object&gt;</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.find_similar_chains" title="Permalink to this definition">¶</a></dt>
<dd><p>Find all chains with at least the specified fractional sequence identity to
the target, and extract as related_chain objects with new pdb hierarchies.</p>
</dd></dl>

<dl class="function">
<dt id="mmtbx.building.make_library.join_hierarchies">
<code class="descclassname">mmtbx.building.make_library.</code><code class="descname">join_hierarchies</code><span class="sig-paren">(</span><em>hierarchies</em>, <em>model_ids=None</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.join_hierarchies" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="mmtbx.building.make_library.load_all_models_in_directory">
<code class="descclassname">mmtbx.building.make_library.</code><code class="descname">load_all_models_in_directory</code><span class="sig-paren">(</span><em>dir_name</em>, <em>limit_extensions=True</em>, <em>recursive=False</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.load_all_models_in_directory" title="Permalink to this definition">¶</a></dt>
<dd><p>Load all models in the specified directory, returning a list of file names
and iotbx.file_reader objects.</p>
</dd></dl>

<dl class="function">
<dt id="mmtbx.building.make_library.load_pdb_models">
<code class="descclassname">mmtbx.building.make_library.</code><code class="descname">load_pdb_models</code><span class="sig-paren">(</span><em>pdb_ids</em>, <em>fault_tolerant=False</em>, <em>log=&lt;libtbx.utils.null_out object&gt;</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.load_pdb_models" title="Permalink to this definition">¶</a></dt>
<dd><p>Given a list of PDB IDs, load hierarchies for all of them (ignoring any
unknown atoms).</p>
</dd></dl>

<dl class="function">
<dt id="mmtbx.building.make_library.load_related_models_from_pdb">
<code class="descclassname">mmtbx.building.make_library.</code><code class="descname">load_related_models_from_pdb</code><span class="sig-paren">(</span><em>pdb_ids</em>, <em>**kwds</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.load_related_models_from_pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Combines load_pdb_models and extract_related_models, with multiprocessing
support.  Functionally equivalent to load_related_models_in_directory but
with models from the PDB (local or remote).</p>
</dd></dl>

<dl class="function">
<dt id="mmtbx.building.make_library.load_related_models_in_directory">
<code class="descclassname">mmtbx.building.make_library.</code><code class="descname">load_related_models_in_directory</code><span class="sig-paren">(</span><em>dir_name</em>, <em>sequence</em>, <em>limit_extensions=True</em>, <em>recursive=False</em>, <em>first_chain_only=False</em>, <em>reset_chain_id=None</em>, <em>min_identity=0.95</em>, <em>return_original_files=False</em>, <em>nproc=&lt;libtbx.AutoType object&gt;</em>, <em>log=&lt;libtbx.utils.null_out object&gt;</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.load_related_models_in_directory" title="Permalink to this definition">¶</a></dt>
<dd><p>Combines load_all_models_in_directory and extract_related_models.</p>
</dd></dl>

<dl class="class">
<dt id="mmtbx.building.make_library.related_chain">
<em class="property">class </em><code class="descclassname">mmtbx.building.make_library.</code><code class="descname">related_chain</code><span class="sig-paren">(</span><em>**kwds</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.related_chain" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="../libtbx/index.html#libtbx.slots_getstate_setstate_default_initializer" title="libtbx.slots_getstate_setstate_default_initializer"><code class="xref py py-class docutils literal"><span class="pre">libtbx.slots_getstate_setstate_default_initializer</span></code></a></p>
<p>Container for a protein chain (as hierarchy object) and metadata, extracted
based on similarity to a target sequence.</p>
<dl class="attribute">
<dt id="mmtbx.building.make_library.related_chain.chain_id">
<code class="descname">chain_id</code><a class="headerlink" href="#mmtbx.building.make_library.related_chain.chain_id" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="mmtbx.building.make_library.related_chain.identity">
<code class="descname">identity</code><a class="headerlink" href="#mmtbx.building.make_library.related_chain.identity" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="mmtbx.building.make_library.related_chain.pdb_hierarchy">
<code class="descname">pdb_hierarchy</code><a class="headerlink" href="#mmtbx.building.make_library.related_chain.pdb_hierarchy" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="mmtbx.building.make_library.related_chain.source_info">
<code class="descname">source_info</code><a class="headerlink" href="#mmtbx.building.make_library.related_chain.source_info" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="function">
<dt id="mmtbx.building.make_library.selection_by_symmetry">
<code class="descclassname">mmtbx.building.make_library.</code><code class="descname">selection_by_symmetry</code><span class="sig-paren">(</span><em>crystal_symmetries</em>, <em>source_infos</em>, <em>exclude_space_group=None</em>, <em>only_space_group=None</em>, <em>log=&lt;libtbx.utils.null_out object&gt;</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.selection_by_symmetry" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a list of indices for those crystal symmetries which either match the
only_space_group parameter, or do not match exclude_space_group.</p>
</dd></dl>

</div>
<div class="section" id="module-mmtbx.building">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-mmtbx.building" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="mmtbx.building.atom_group_as_hierarchy">
<code class="descclassname">mmtbx.building.</code><code class="descname">atom_group_as_hierarchy</code><span class="sig-paren">(</span><em>atom_group</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.atom_group_as_hierarchy" title="Permalink to this definition">¶</a></dt>
<dd><p>Convert an iotbx.pdb.hierarchy.atom_group object to a separate hierarchy
(leaving the original hierarchy untouched), which allows us to use the
pickling capabilities of iotbx.pdb.hierarchy.root.</p>
</dd></dl>

<dl class="class">
<dt id="mmtbx.building.box_build_refine_base">
<em class="property">class </em><code class="descclassname">mmtbx.building.</code><code class="descname">box_build_refine_base</code><span class="sig-paren">(</span><em>pdb_hierarchy</em>, <em>xray_structure</em>, <em>processed_pdb_file</em>, <em>target_map</em>, <em>selection</em>, <em>d_min</em>, <em>out</em>, <em>cif_objects=()</em>, <em>resolution_factor=0.25</em>, <em>selection_buffer_radius=5</em>, <em>box_cushion=2</em>, <em>target_map_rsr=None</em>, <em>geometry_restraints_manager=None</em>, <em>debug=False</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/2/library/functions.html#object" title="(in Python v2.7)"><code class="xref py py-class docutils literal"><span class="pre">object</span></code></a></p>
<p>Base class for handling functions associated with rebuilding and refinement
of atoms in a box, with the building methods implemented elsewhere.</p>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.box_build_refine_base.anneal" title="mmtbx.building.box_build_refine_base.anneal"><code class="xref py py-obj docutils literal"><span class="pre">anneal</span></code></a>([simulated_annealing_params,&nbsp;...])</td>
<td>Run real-space simulated annealing using the target map (not the RSR map, if this is different).</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mmtbx.building.box_build_refine_base.cc_model_map" title="mmtbx.building.box_build_refine_base.cc_model_map"><code class="xref py py-obj docutils literal"><span class="pre">cc_model_map</span></code></a>([selection,&nbsp;radius])</td>
<td>Calculate the correlation coefficient for the current model (in terms of F(calc) from the xray structure) and the target map, calculated at atomic positions rather than grid points.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.box_build_refine_base.fit_residue_in_box" title="mmtbx.building.box_build_refine_base.fit_residue_in_box"><code class="xref py py-obj docutils literal"><span class="pre">fit_residue_in_box</span></code></a>([mon_lib_srv,&nbsp;...])</td>
<td></td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mmtbx.building.box_build_refine_base.geometry_minimization" title="mmtbx.building.box_build_refine_base.geometry_minimization"><code class="xref py py-obj docutils literal"><span class="pre">geometry_minimization</span></code></a>([selection,&nbsp;bond,&nbsp;...])</td>
<td>Perform geometry minimization on a selection of boxed atoms.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.box_build_refine_base.get_selected_sites" title="mmtbx.building.box_build_refine_base.get_selected_sites"><code class="xref py py-obj docutils literal"><span class="pre">get_selected_sites</span></code></a>([selection,&nbsp;hydrogens])</td>
<td></td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mmtbx.building.box_build_refine_base.mean_density_at_sites" title="mmtbx.building.box_build_refine_base.mean_density_at_sites"><code class="xref py py-obj docutils literal"><span class="pre">mean_density_at_sites</span></code></a>([selection])</td>
<td>Compute the mean density of the target map at coordinates for non-hydrogen atoms.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.box_build_refine_base.only_residue" title="mmtbx.building.box_build_refine_base.only_residue"><code class="xref py py-obj docutils literal"><span class="pre">only_residue</span></code></a>()</td>
<td>Returns the atom_group object corresponding to the original selection.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mmtbx.building.box_build_refine_base.real_space_refine" title="mmtbx.building.box_build_refine_base.real_space_refine"><code class="xref py py-obj docutils literal"><span class="pre">real_space_refine</span></code></a>([selection])</td>
<td>Run simple real-space refinement, returning the optimized weight for the map target (wx).</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.box_build_refine_base.restrain_atoms" title="mmtbx.building.box_build_refine_base.restrain_atoms"><code class="xref py py-obj docutils literal"><span class="pre">restrain_atoms</span></code></a>(selection,&nbsp;reference_sigma[,&nbsp;...])</td>
<td>Apply harmonic reference restraints to the selected atoms, wiping out any previously existing harmonic restraints.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mmtbx.building.box_build_refine_base.update_coordinates" title="mmtbx.building.box_build_refine_base.update_coordinates"><code class="xref py py-obj docutils literal"><span class="pre">update_coordinates</span></code></a>(sites_cart)</td>
<td>Set coordinates for the boxed xray_structure and pdb_hierarchy objects.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.box_build_refine_base.update_original_coordinates" title="mmtbx.building.box_build_refine_base.update_original_coordinates"><code class="xref py py-obj docutils literal"><span class="pre">update_original_coordinates</span></code></a>()</td>
<td>Copies the new coordinates from the box to the selected atoms in the original xray_structure object, and returns the (complete) list of coordinates.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mmtbx.building.box_build_refine_base.update_sites_from_pdb_atoms" title="mmtbx.building.box_build_refine_base.update_sites_from_pdb_atoms"><code class="xref py py-obj docutils literal"><span class="pre">update_sites_from_pdb_atoms</span></code></a>()</td>
<td>Update the xray_structure coordinates after manipulating the PDB hierarchy.</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="mmtbx.building.box_build_refine_base.anneal">
<code class="descname">anneal</code><span class="sig-paren">(</span><em>simulated_annealing_params=None</em>, <em>start_temperature=None</em>, <em>cool_rate=None</em>, <em>number_of_steps=50</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.anneal" title="Permalink to this definition">¶</a></dt>
<dd><p>Run real-space simulated annealing using the target map (not the RSR map,
if this is different).  In practice, the non-selection atoms in the box
should almost always be restrained to their current positions, but the
setup is left to the calling code.</p>
</dd></dl>

<dl class="method">
<dt id="mmtbx.building.box_build_refine_base.cc_model_map">
<code class="descname">cc_model_map</code><span class="sig-paren">(</span><em>selection=None</em>, <em>radius=1.5</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.cc_model_map" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate the correlation coefficient for the current model (in terms of
F(calc) from the xray structure) and the target map, calculated at atomic
positions rather than grid points.  This will be much
less accurate than the CC calculated in the original crystal environment,
with full F(model) including bulk solvent correction.</p>
</dd></dl>

<dl class="method">
<dt id="mmtbx.building.box_build_refine_base.fit_residue_in_box">
<code class="descname">fit_residue_in_box</code><span class="sig-paren">(</span><em>mon_lib_srv=None</em>, <em>rotamer_manager=None</em>, <em>backbone_sample_angle=20</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.fit_residue_in_box" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.box_build_refine_base.geometry_minimization">
<code class="descname">geometry_minimization</code><span class="sig-paren">(</span><em>selection=None</em>, <em>bond=True</em>, <em>nonbonded=True</em>, <em>angle=True</em>, <em>dihedral=True</em>, <em>chirality=True</em>, <em>planarity=True</em>, <em>max_number_of_iterations=500</em>, <em>number_of_macro_cycles=5</em>, <em>rmsd_bonds_termination_cutoff=0</em>, <em>rmsd_angles_termination_cutoff=0</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.geometry_minimization" title="Permalink to this definition">¶</a></dt>
<dd><p>Perform geometry minimization on a selection of boxed atoms.</p>
</dd></dl>

<dl class="method">
<dt id="mmtbx.building.box_build_refine_base.get_selected_sites">
<code class="descname">get_selected_sites</code><span class="sig-paren">(</span><em>selection=None</em>, <em>hydrogens=True</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.get_selected_sites" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.box_build_refine_base.mean_density_at_sites">
<code class="descname">mean_density_at_sites</code><span class="sig-paren">(</span><em>selection=None</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.mean_density_at_sites" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute the mean density of the target map at coordinates for non-hydrogen
atoms.</p>
</dd></dl>

<dl class="method">
<dt id="mmtbx.building.box_build_refine_base.only_residue">
<code class="descname">only_residue</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.only_residue" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the atom_group object corresponding to the original selection.
Will raise an exception if the selection covers more than one atom_group.</p>
</dd></dl>

<dl class="method">
<dt id="mmtbx.building.box_build_refine_base.real_space_refine">
<code class="descname">real_space_refine</code><span class="sig-paren">(</span><em>selection=None</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.real_space_refine" title="Permalink to this definition">¶</a></dt>
<dd><p>Run simple real-space refinement, returning the optimized weight for the
map target (wx).</p>
</dd></dl>

<dl class="method">
<dt id="mmtbx.building.box_build_refine_base.restrain_atoms">
<code class="descname">restrain_atoms</code><span class="sig-paren">(</span><em>selection</em>, <em>reference_sigma</em>, <em>limit=1.0</em>, <em>reset_first=True</em>, <em>top_out_potential=False</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.restrain_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply harmonic reference restraints to the selected atoms, wiping out
any previously existing harmonic restraints.</p>
</dd></dl>

<dl class="method">
<dt id="mmtbx.building.box_build_refine_base.update_coordinates">
<code class="descname">update_coordinates</code><span class="sig-paren">(</span><em>sites_cart</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.update_coordinates" title="Permalink to this definition">¶</a></dt>
<dd><p>Set coordinates for the boxed xray_structure and pdb_hierarchy objects.</p>
</dd></dl>

<dl class="method">
<dt id="mmtbx.building.box_build_refine_base.update_original_coordinates">
<code class="descname">update_original_coordinates</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.update_original_coordinates" title="Permalink to this definition">¶</a></dt>
<dd><p>Copies the new coordinates from the box to the selected atoms in the
original xray_structure object, and returns the (complete) list of
coordinates.</p>
</dd></dl>

<dl class="method">
<dt id="mmtbx.building.box_build_refine_base.update_sites_from_pdb_atoms">
<code class="descname">update_sites_from_pdb_atoms</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.update_sites_from_pdb_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Update the xray_structure coordinates after manipulating the PDB hierarchy.</p>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="mmtbx.building.extract_iselection">
<code class="descclassname">mmtbx.building.</code><code class="descname">extract_iselection</code><span class="sig-paren">(</span><em>pdb_objects</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extract_iselection" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="mmtbx.building.generate_sidechain_clusters">
<code class="descclassname">mmtbx.building.</code><code class="descname">generate_sidechain_clusters</code><span class="sig-paren">(</span><em>residue</em>, <em>mon_lib_srv</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.generate_sidechain_clusters" title="Permalink to this definition">¶</a></dt>
<dd><p>Extract Chi angle indices (including rotation axis) from the atom_group</p>
</dd></dl>

<dl class="function">
<dt id="mmtbx.building.get_difference_maps">
<code class="descclassname">mmtbx.building.</code><code class="descname">get_difference_maps</code><span class="sig-paren">(</span><em>fmodel</em>, <em>resolution_factor=0.25</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.get_difference_maps" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="mmtbx.building.get_model_map_stats">
<code class="descclassname">mmtbx.building.</code><code class="descname">get_model_map_stats</code><span class="sig-paren">(</span><em>selection</em>, <em>target_map</em>, <em>model_map</em>, <em>unit_cell</em>, <em>sites_cart</em>, <em>pdb_atoms</em>, <em>local_sampling=False</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.get_model_map_stats" title="Permalink to this definition">¶</a></dt>
<dd><p>Collect basic statistics for a model map and some target map (usually an
mFo-DFc map), including CC, mean, and minimum density at the atomic
positions.</p>
</dd></dl>

<dl class="function">
<dt id="mmtbx.building.get_nearby_water_selection">
<code class="descclassname">mmtbx.building.</code><code class="descname">get_nearby_water_selection</code><span class="sig-paren">(</span><em>pdb_hierarchy</em>, <em>xray_structure</em>, <em>selection</em>, <em>selection_buffer_radius=5</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.get_nearby_water_selection" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="mmtbx.building.get_non_hydrogen_atom_indices">
<code class="descclassname">mmtbx.building.</code><code class="descname">get_non_hydrogen_atom_indices</code><span class="sig-paren">(</span><em>pdb_object</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.get_non_hydrogen_atom_indices" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="mmtbx.building.get_window_around_residue">
<code class="descclassname">mmtbx.building.</code><code class="descname">get_window_around_residue</code><span class="sig-paren">(</span><em>residue</em>, <em>window_size=2</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.get_window_around_residue" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="mmtbx.building.is_stub_residue">
<code class="descclassname">mmtbx.building.</code><code class="descname">is_stub_residue</code><span class="sig-paren">(</span><em>atom_group</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.is_stub_residue" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="mmtbx.building.iter_residue_groups">
<code class="descclassname">mmtbx.building.</code><code class="descname">iter_residue_groups</code><span class="sig-paren">(</span><em>hierarchy</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.iter_residue_groups" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="mmtbx.building.local_density_quality">
<em class="property">class </em><code class="descclassname">mmtbx.building.</code><code class="descname">local_density_quality</code><span class="sig-paren">(</span><em>fofc_map</em>, <em>two_fofc_map</em>, <em>sites_cart=None</em>, <em>atom_names=None</em>, <em>unit_cell=None</em>, <em>atom_selection=None</em>, <em>xray_structure=None</em>, <em>radius=2.5</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/2/library/functions.html#object" title="(in Python v2.7)"><code class="xref py py-class docutils literal"><span class="pre">object</span></code></a></p>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.local_density_quality.atoms_below_fofc_map_level" title="mmtbx.building.local_density_quality.atoms_below_fofc_map_level"><code class="xref py py-obj docutils literal"><span class="pre">atoms_below_fofc_map_level</span></code></a>([sigma_cutoff])</td>
<td></td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mmtbx.building.local_density_quality.atoms_below_two_fofc_map_level" title="mmtbx.building.local_density_quality.atoms_below_two_fofc_map_level"><code class="xref py py-obj docutils literal"><span class="pre">atoms_below_two_fofc_map_level</span></code></a>([sigma_cutoff])</td>
<td></td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.local_density_quality.atoms_outside_density" title="mmtbx.building.local_density_quality.atoms_outside_density"><code class="xref py py-obj docutils literal"><span class="pre">atoms_outside_density</span></code></a>(**kwds)</td>
<td></td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mmtbx.building.local_density_quality.density_at_atom" title="mmtbx.building.local_density_quality.density_at_atom"><code class="xref py py-obj docutils literal"><span class="pre">density_at_atom</span></code></a>(atom_name)</td>
<td>Return a simple object containing density levels for the named atom, or None if no match was found.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.local_density_quality.fraction_of_nearby_grid_points_above_cutoff" title="mmtbx.building.local_density_quality.fraction_of_nearby_grid_points_above_cutoff"><code class="xref py py-obj docutils literal"><span class="pre">fraction_of_nearby_grid_points_above_cutoff</span></code></a>([...])</td>
<td></td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mmtbx.building.local_density_quality.max_fofc_value" title="mmtbx.building.local_density_quality.max_fofc_value"><code class="xref py py-obj docutils literal"><span class="pre">max_fofc_value</span></code></a>()</td>
<td></td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.local_density_quality.number_of_atoms_below_fofc_map_level" title="mmtbx.building.local_density_quality.number_of_atoms_below_fofc_map_level"><code class="xref py py-obj docutils literal"><span class="pre">number_of_atoms_below_fofc_map_level</span></code></a>(*args,&nbsp;...)</td>
<td></td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mmtbx.building.local_density_quality.number_of_atoms_below_two_fofc_map_level" title="mmtbx.building.local_density_quality.number_of_atoms_below_two_fofc_map_level"><code class="xref py py-obj docutils literal"><span class="pre">number_of_atoms_below_two_fofc_map_level</span></code></a>(...)</td>
<td></td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mmtbx.building.local_density_quality.number_of_atoms_outside_density" title="mmtbx.building.local_density_quality.number_of_atoms_outside_density"><code class="xref py py-obj docutils literal"><span class="pre">number_of_atoms_outside_density</span></code></a>(*args,&nbsp;**kwds)</td>
<td></td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mmtbx.building.local_density_quality.show_atoms_outside_density" title="mmtbx.building.local_density_quality.show_atoms_outside_density"><code class="xref py py-obj docutils literal"><span class="pre">show_atoms_outside_density</span></code></a>(**kwds)</td>
<td></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="mmtbx.building.local_density_quality.atoms_below_fofc_map_level">
<code class="descname">atoms_below_fofc_map_level</code><span class="sig-paren">(</span><em>sigma_cutoff=-2.5</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.atoms_below_fofc_map_level" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.local_density_quality.atoms_below_two_fofc_map_level">
<code class="descname">atoms_below_two_fofc_map_level</code><span class="sig-paren">(</span><em>sigma_cutoff=-2.5</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.atoms_below_two_fofc_map_level" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.local_density_quality.atoms_outside_density">
<code class="descname">atoms_outside_density</code><span class="sig-paren">(</span><em>**kwds</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.atoms_outside_density" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.local_density_quality.density_at_atom">
<code class="descname">density_at_atom</code><span class="sig-paren">(</span><em>atom_name</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.density_at_atom" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a simple object containing density levels for the named atom,
or None if no match was found.</p>
</dd></dl>

<dl class="method">
<dt id="mmtbx.building.local_density_quality.fraction_of_nearby_grid_points_above_cutoff">
<code class="descname">fraction_of_nearby_grid_points_above_cutoff</code><span class="sig-paren">(</span><em>sigma_cutoff=2.5</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.fraction_of_nearby_grid_points_above_cutoff" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.local_density_quality.max_fofc_value">
<code class="descname">max_fofc_value</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.max_fofc_value" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.local_density_quality.number_of_atoms_below_fofc_map_level">
<code class="descname">number_of_atoms_below_fofc_map_level</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwds</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.number_of_atoms_below_fofc_map_level" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.local_density_quality.number_of_atoms_below_two_fofc_map_level">
<code class="descname">number_of_atoms_below_two_fofc_map_level</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwds</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.number_of_atoms_below_two_fofc_map_level" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.local_density_quality.number_of_atoms_outside_density">
<code class="descname">number_of_atoms_outside_density</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwds</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.number_of_atoms_outside_density" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="mmtbx.building.local_density_quality.show_atoms_outside_density">
<code class="descname">show_atoms_outside_density</code><span class="sig-paren">(</span><em>**kwds</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.show_atoms_outside_density" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="function">
<dt id="mmtbx.building.remove_sidechain_atoms">
<code class="descclassname">mmtbx.building.</code><code class="descname">remove_sidechain_atoms</code><span class="sig-paren">(</span><em>pdb_objects</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.remove_sidechain_atoms" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="mmtbx.building.reprocess_pdb">
<code class="descclassname">mmtbx.building.</code><code class="descname">reprocess_pdb</code><span class="sig-paren">(</span><em>pdb_hierarchy</em>, <em>crystal_symmetry</em>, <em>cif_objects</em>, <em>out</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.reprocess_pdb" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="mmtbx.building.residue_density_quality">
<code class="descclassname">mmtbx.building.</code><code class="descname">residue_density_quality</code><span class="sig-paren">(</span><em>atom_group</em>, <em>unit_cell</em>, <em>two_fofc_map</em>, <em>fofc_map</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.residue_density_quality" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="mmtbx.building.residues_are_adjacent">
<code class="descclassname">mmtbx.building.</code><code class="descname">residues_are_adjacent</code><span class="sig-paren">(</span><em>residue1</em>, <em>residue2</em>, <em>max_sep=2.5</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.residues_are_adjacent" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="mmtbx.building.run_real_space_annealing">
<code class="descclassname">mmtbx.building.</code><code class="descname">run_real_space_annealing</code><span class="sig-paren">(</span><em>xray_structure</em>, <em>pdb_hierarchy</em>, <em>selection</em>, <em>target_map</em>, <em>d_min</em>, <em>processed_pdb_file=None</em>, <em>cif_objects=()</em>, <em>resolution_factor=0.25</em>, <em>params=None</em>, <em>wc=1</em>, <em>target_map_rsr=None</em>, <em>rsr_after_anneal=False</em>, <em>reference_sigma=0.5</em>, <em>out=None</em>, <em>debug=False</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.run_real_space_annealing" title="Permalink to this definition">¶</a></dt>
<dd><p>Run simulated annealing with a real-space target.  For maximum flexibility,
a separate map may be used for the initial real-space refinement step.</p>
</dd></dl>

<dl class="function">
<dt id="mmtbx.building.show_chain_resseq_ranges">
<code class="descclassname">mmtbx.building.</code><code class="descname">show_chain_resseq_ranges</code><span class="sig-paren">(</span><em>residues</em>, <em>out=&lt;open file '&lt;stdout&gt;'</em>, <em>mode 'w'&gt;</em>, <em>prefix=''</em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.show_chain_resseq_ranges" title="Permalink to this definition">¶</a></dt>
<dd><p>Given a list of residues (either residue_group or atom_group objects),
print a summary of the chains and residue ranges present.</p>
</dd></dl>

</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="../index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">mmtbx.building package</a><ul>
<li><a class="reference internal" href="#subpackages">Subpackages</a></li>
<li><a class="reference internal" href="#submodules">Submodules</a></li>
<li><a class="reference internal" href="#module-mmtbx.building.extend_sidechains">mmtbx.building.extend_sidechains module</a><ul>
</ul>
</li>
<li><a class="reference internal" href="#module-mmtbx.building.make_library">mmtbx.building.make_library module</a><ul>
</ul>
</li>
<li><a class="reference internal" href="#module-mmtbx.building">Module contents</a><ul>
</ul>
</li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="mmtbx.html"
                        title="previous chapter">mmtbx - macromolecular crystallography</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="mmtbx.building.alternate_conformations.html"
                        title="next chapter">mmtbx.building.alternate_conformations package</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/mmtbx/mmtbx.building.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="../py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="mmtbx.building.alternate_conformations.html" title="mmtbx.building.alternate_conformations package"
             >next</a> |</li>
        <li class="right" >
          <a href="mmtbx.html" title="mmtbx - macromolecular crystallography"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">CCTBX None documentation</a> &raquo;</li>
          <li class="nav-item nav-item-1"><a href="mmtbx.html" >mmtbx - macromolecular crystallography</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &copy; Copyright 2014, University of California.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.4.
    </div>
  </body>
</html>