<!DOCTYPE html>

<html lang="en" data-content_root="../">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>mmtbx.building package &#8212; CCTBX Developer documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../_static/classic.css?v=514cf933" />
    
    <script src="../_static/documentation_options.js?v=9c77bcdb"></script>
    <script src="../_static/doctools.js?v=888ff710"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="mmtbx.bulk_solvent package" href="mmtbx.bulk_solvent.html" />
    <link rel="prev" title="mmtbx - macromolecular crystallography" href="mmtbx.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="../py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="mmtbx.bulk_solvent.html" title="mmtbx.bulk_solvent package"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="mmtbx.html" title="mmtbx - macromolecular crystallography"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">CCTBX Developer documentation</a> &#187;</li>
          <li class="nav-item nav-item-1"><a href="mmtbx.html" accesskey="U">mmtbx - macromolecular crystallography</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">mmtbx.building package</a></li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="mmtbx-building-package">
<h1>mmtbx.building package<a class="headerlink" href="#mmtbx-building-package" title="Link to this heading">¶</a></h1>
<section id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Link to this heading">¶</a></h2>
</section>
<section id="module-mmtbx.building.extend_sidechains">
<span id="mmtbx-building-extend-sidechains-module"></span><h2>mmtbx.building.extend_sidechains module<a class="headerlink" href="#module-mmtbx.building.extend_sidechains" title="Link to this heading">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="mmtbx.building.extend_sidechains.conformation_scorer">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">mmtbx.building.extend_sidechains.</span></span><span class="sig-name descname"><span class="pre">conformation_scorer</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">old_residue</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">new_residue</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.conformation_scorer" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.11)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>Stand-in for the conformation scoring class in mmtbx.refinement.real_space;
instead of calculating fit to the map, this simply uses the change in
position of the first atom being moved at each rotation.  This allows us to
superimpose the conformations for those atoms which are present in both the
old and the new residues.</p>
<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.extend_sidechains.conformation_scorer.apply_final">
<span class="sig-name descname"><span class="pre">apply_final</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.conformation_scorer.apply_final" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.extend_sidechains.conformation_scorer.reset">
<span class="sig-name descname"><span class="pre">reset</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sites_cart</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">selection</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.conformation_scorer.reset" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.extend_sidechains.conformation_scorer.update">
<span class="sig-name descname"><span class="pre">update</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sites_cart</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">selection</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.conformation_scorer.update" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.extend_sidechains.correct_sequence">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.extend_sidechains.</span></span><span class="sig-name descname"><span class="pre">correct_sequence</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_hierarchy</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sequences</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">truncate_to_cbeta=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out=&lt;colorama.ansitowin32.StreamWrapper</span> <span class="pre">object&gt;</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.correct_sequence" title="Link to this definition">¶</a></dt>
<dd><p>Modify the sequence for the pdb hierarchy to match that of the aligned
sequence.  This will remove incompatible atoms; the sidechains will still
need to be extended separated.  For proteins only - mismatches in nucleic
acids will only result in a warning.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pdb_hierarchy</strong> – iotbx.pdb.hierarchy.root object</p></li>
<li><p><strong>sequences</strong> – list of iotbx.bioinformatics.sequence objects</p></li>
<li><p><strong>trucate_to_cbeta</strong> – chop off entire sidechain to C-beta (default: leave
common atoms in place)</p></li>
<li><p><strong>out</strong> – output filehandle (default = stdout)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>number of atom_group objects renamed</p>
</dd>
</dl>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="mmtbx.building.extend_sidechains.extend_and_refine">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">mmtbx.building.extend_sidechains.</span></span><span class="sig-name descname"><span class="pre">extend_and_refine</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_hierarchy</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xray_structure</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fmodel</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">params</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cif_objects=()</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out=&lt;colorama.ansitowin32.StreamWrapper</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_model=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_map_coeffs=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">prefix=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">write_files=True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reset_segid=True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose=True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.extend_and_refine" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.11)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>Run the combined sidechain extension and real-space fitting, and optionally
write final model and map coefficients.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.extend_sidechains.extend_protein_model">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.extend_sidechains.</span></span><span class="sig-name descname"><span class="pre">extend_protein_model</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_hierarchy</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mon_lib_srv</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">add_hydrogens</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">selection</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.extend_protein_model" title="Link to this definition">¶</a></dt>
<dd><p>Rebuild a sidechain by substituting an ideal amino acid and rotating the
sidechain to match the old conformation as closely as possible.
Limited functionality:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>Amino-acids only,</p></li>
<li><p>side chain atoms only.</p></li>
<li><p>Not terminii aware.</p></li>
<li><p>Not aware of v2.3 vs v3.2 atom names e.g. HB1,HB2 vs HB2,HB3.</p></li>
<li><p>Skips altlocs.</p></li>
</ol>
</div></blockquote>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.extend_sidechains.extend_residue">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.extend_sidechains.</span></span><span class="sig-name descname"><span class="pre">extend_residue</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">residue</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_atom_group</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mon_lib_srv</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.extend_residue" title="Link to this definition">¶</a></dt>
<dd><p>Rebuild a sidechain by substituting an ideal amino acid and rotating the
sidechain to match the old conformation as closely as possible.
Limited functionality:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>Amino-acids only, 2) side chain atoms only.</p></li>
</ol>
</div></blockquote>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="mmtbx.building.extend_sidechains.prefilter">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">mmtbx.building.extend_sidechains.</span></span><span class="sig-name descname"><span class="pre">prefilter</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fmodel</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">backbone_min_sigma</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.0</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.prefilter" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.11)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>Optional filter for excluding residues with poor backbone density from being
extended.  This is done as a separate callback to enable the main rebuilding
routine to be independent of data/maps.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.extend_sidechains.refit_residues">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.extend_sidechains.</span></span><span class="sig-name descname"><span class="pre">refit_residues</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_hierarchy</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cif_objects</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fmodel</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">use_rotamers=True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">anneal=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose=True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">allow_modified_residues=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out=&lt;colorama.ansitowin32.StreamWrapper</span> <span class="pre">object&gt;</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extend_sidechains.refit_residues" title="Link to this definition">¶</a></dt>
<dd><p>Use real-space refinement tools to fit newly extended residues.</p>
</dd></dl>

</section>
<section id="module-mmtbx.building.make_library">
<span id="mmtbx-building-make-library-module"></span><h2>mmtbx.building.make_library module<a class="headerlink" href="#module-mmtbx.building.make_library" title="Link to this heading">¶</a></h2>
<p>Tools for assembling an ensemble of related structures for local rebuilding by
homology.</p>
<dl class="py class">
<dt class="sig sig-object py" id="mmtbx.building.make_library.ensembler">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">mmtbx.building.make_library.</span></span><span class="sig-name descname"><span class="pre">ensembler</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">reference_hierarchy</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">related_chains</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atom_selection_string=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">backbone_only=&lt;libtbx.AutoType</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">calpha_only=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nproc=&lt;libtbx.AutoType</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sieve_fit=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">frac_discard=0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">log=&lt;libtbx.utils.null_out</span> <span class="pre">object&gt;</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.ensembler" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.11)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>Superpose a collection of single-chain models on a reference chain.  This is
intentionally very limited in function, but flexible with respect to the
choice of atoms to superpose.</p>
<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.make_library.ensembler.align_model">
<span class="sig-name descname"><span class="pre">align_model</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">i_model</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.ensembler.align_model" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.make_library.ensembler.as_multi_model_hierarchy">
<span class="sig-name descname"><span class="pre">as_multi_model_hierarchy</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.ensembler.as_multi_model_hierarchy" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.make_library.ensembler.realign">
<span class="sig-name descname"><span class="pre">realign</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atom_selection_string</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.ensembler.realign" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.make_library.extract_and_superpose">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.make_library.</span></span><span class="sig-name descname"><span class="pre">extract_and_superpose</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">reference_hierarchy</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">search_directory</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sequence</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">params</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nproc=&lt;libtbx.AutoType</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out=&lt;colorama.ansitowin32.StreamWrapper</span> <span class="pre">object&gt;</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.extract_and_superpose" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.make_library.extract_peptide_fragments_by_sequence">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.make_library.</span></span><span class="sig-name descname"><span class="pre">extract_peptide_fragments_by_sequence</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_hierarchy</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sequence</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">renumber_from</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mainchain_only</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remove_hydrogens</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reset_chain_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'A'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reset_icode</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.extract_peptide_fragments_by_sequence" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="mmtbx.building.make_library.extract_related_models">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">mmtbx.building.make_library.</span></span><span class="sig-name descname"><span class="pre">extract_related_models</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sources_and_models</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sequence</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">first_chain_only=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reset_chain_id=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_identity=0.95</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nproc=&lt;libtbx.AutoType</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">log=&lt;libtbx.utils.null_out</span> <span class="pre">object&gt;</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.extract_related_models" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.11)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>Multiprocessing wrapper for pulling matching chains out of a collection of
PDB hierarchies (either from local files or from a PDB mirror).</p>
<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.make_library.extract_related_models.examine_model">
<span class="sig-name descname"><span class="pre">examine_model</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">i_model</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.extract_related_models.examine_model" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.make_library.extract_related_models.results">
<span class="sig-name descname"><span class="pre">results</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.extract_related_models.results" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.make_library.fetch_similar_pdb_ids">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.make_library.</span></span><span class="sig-name descname"><span class="pre">fetch_similar_pdb_ids</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sequence</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_identity</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.95</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_resolution</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xray_only</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">data_only</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">expect</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1e-07</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sort_by_resolution</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ligands</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">log</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.fetch_similar_pdb_ids" title="Link to this definition">¶</a></dt>
<dd><p>Finds closely related structures in the RCSB database, optionally limited to
structures with specific ligands.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.make_library.find_similar_chains">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.make_library.</span></span><span class="sig-name descname"><span class="pre">find_similar_chains</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_hierarchy</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sequence</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">source_info=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">first_chain_only=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remove_alt_confs=True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reset_chain_id=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_identity=0.95</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_identity_epsilon=0.02</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">log=&lt;libtbx.utils.null_out</span> <span class="pre">object&gt;</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.find_similar_chains" title="Link to this definition">¶</a></dt>
<dd><p>Find all chains with at least the specified fractional sequence identity to
the target, and extract as related_chain objects with new pdb hierarchies.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.make_library.join_hierarchies">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.make_library.</span></span><span class="sig-name descname"><span class="pre">join_hierarchies</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">hierarchies</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_ids</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.join_hierarchies" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.make_library.load_all_models_in_directory">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.make_library.</span></span><span class="sig-name descname"><span class="pre">load_all_models_in_directory</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dir_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">limit_extensions</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recursive</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.load_all_models_in_directory" title="Link to this definition">¶</a></dt>
<dd><p>Load all models in the specified directory, returning a list of file names
and iotbx.file_reader objects.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.make_library.load_pdb_models">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.make_library.</span></span><span class="sig-name descname"><span class="pre">load_pdb_models</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_ids</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fault_tolerant=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">log=&lt;libtbx.utils.null_out</span> <span class="pre">object&gt;</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.load_pdb_models" title="Link to this definition">¶</a></dt>
<dd><p>Given a list of PDB IDs, load hierarchies for all of them (ignoring any
unknown atoms).</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.make_library.load_related_models_from_pdb">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.make_library.</span></span><span class="sig-name descname"><span class="pre">load_related_models_from_pdb</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_ids</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwds</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.load_related_models_from_pdb" title="Link to this definition">¶</a></dt>
<dd><p>Combines load_pdb_models and extract_related_models, with multiprocessing
support.  Functionally equivalent to load_related_models_in_directory but
with models from the PDB (local or remote).</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.make_library.load_related_models_in_directory">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.make_library.</span></span><span class="sig-name descname"><span class="pre">load_related_models_in_directory</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dir_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sequence</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">limit_extensions=True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recursive=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">first_chain_only=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reset_chain_id=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_identity=0.95</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_original_files=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nproc=&lt;libtbx.AutoType</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">log=&lt;libtbx.utils.null_out</span> <span class="pre">object&gt;</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.load_related_models_in_directory" title="Link to this definition">¶</a></dt>
<dd><p>Combines load_all_models_in_directory and extract_related_models.</p>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="mmtbx.building.make_library.related_chain">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">mmtbx.building.make_library.</span></span><span class="sig-name descname"><span class="pre">related_chain</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwds</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.related_chain" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="../libtbx/index.html#libtbx.slots_getstate_setstate_default_initializer" title="libtbx.slots_getstate_setstate_default_initializer"><code class="xref py py-class docutils literal notranslate"><span class="pre">slots_getstate_setstate_default_initializer</span></code></a></p>
<p>Container for a protein chain (as hierarchy object) and metadata, extracted
based on similarity to a target sequence.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="mmtbx.building.make_library.related_chain.chain_id">
<span class="sig-name descname"><span class="pre">chain_id</span></span><a class="headerlink" href="#mmtbx.building.make_library.related_chain.chain_id" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="mmtbx.building.make_library.related_chain.identity">
<span class="sig-name descname"><span class="pre">identity</span></span><a class="headerlink" href="#mmtbx.building.make_library.related_chain.identity" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="mmtbx.building.make_library.related_chain.pdb_hierarchy">
<span class="sig-name descname"><span class="pre">pdb_hierarchy</span></span><a class="headerlink" href="#mmtbx.building.make_library.related_chain.pdb_hierarchy" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="mmtbx.building.make_library.related_chain.source_info">
<span class="sig-name descname"><span class="pre">source_info</span></span><a class="headerlink" href="#mmtbx.building.make_library.related_chain.source_info" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.make_library.selection_by_symmetry">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.make_library.</span></span><span class="sig-name descname"><span class="pre">selection_by_symmetry</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">crystal_symmetries</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">source_infos</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">exclude_space_group=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">only_space_group=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">log=&lt;libtbx.utils.null_out</span> <span class="pre">object&gt;</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.make_library.selection_by_symmetry" title="Link to this definition">¶</a></dt>
<dd><p>Return a list of indices for those crystal symmetries which either match the
only_space_group parameter, or do not match exclude_space_group.</p>
</dd></dl>

</section>
<section id="module-mmtbx.building">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-mmtbx.building" title="Link to this heading">¶</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.atom_group_as_hierarchy">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">atom_group_as_hierarchy</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atom_group</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.atom_group_as_hierarchy" title="Link to this definition">¶</a></dt>
<dd><p>Convert an iotbx.pdb.hierarchy.atom_group object to a separate hierarchy
(leaving the original hierarchy untouched), which allows us to use the
pickling capabilities of iotbx.pdb.hierarchy.root.</p>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="mmtbx.building.box_build_refine_base">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">box_build_refine_base</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_hierarchy</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xray_structure</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">processed_pdb_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_map</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">selection</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">d_min</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cif_objects</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">()</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">resolution_factor</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.25</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">selection_buffer_radius</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">box_cushion</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_map_rsr</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geometry_restraints_manager</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">debug</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.11)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>Base class for handling functions associated with rebuilding and refinement
of atoms in a box, with the building methods implemented elsewhere.</p>
<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.box_build_refine_base.anneal">
<span class="sig-name descname"><span class="pre">anneal</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">simulated_annealing_params</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">start_temperature</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cool_rate</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">number_of_steps</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.anneal" title="Link to this definition">¶</a></dt>
<dd><p>Run real-space simulated annealing using the target map (not the RSR map,
if this is different).  In practice, the non-selection atoms in the box
should almost always be restrained to their current positions, but the
setup is left to the calling code.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.box_build_refine_base.cc_model_map">
<span class="sig-name descname"><span class="pre">cc_model_map</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">selection</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radius</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.5</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.cc_model_map" title="Link to this definition">¶</a></dt>
<dd><p>Calculate the correlation coefficient for the current model (in terms of
F(calc) from the xray structure) and the target map, calculated at atomic
positions rather than grid points.  This will be much
less accurate than the CC calculated in the original crystal environment,
with full F(model) including bulk solvent correction.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.box_build_refine_base.fit_residue_in_box">
<span class="sig-name descname"><span class="pre">fit_residue_in_box</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mon_lib_srv</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rotamer_manager</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">backbone_sample_angle</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">20</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.fit_residue_in_box" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.box_build_refine_base.geometry_minimization">
<span class="sig-name descname"><span class="pre">geometry_minimization</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">selection</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bond</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nonbonded</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">angle</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dihedral</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chirality</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">planarity</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_number_of_iterations</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">500</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">number_of_macro_cycles</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rmsd_bonds_termination_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rmsd_angles_termination_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.geometry_minimization" title="Link to this definition">¶</a></dt>
<dd><p>Perform geometry minimization on a selection of boxed atoms.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.box_build_refine_base.get_selected_sites">
<span class="sig-name descname"><span class="pre">get_selected_sites</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">selection</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">hydrogens</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.get_selected_sites" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.box_build_refine_base.mean_density_at_sites">
<span class="sig-name descname"><span class="pre">mean_density_at_sites</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">selection</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.mean_density_at_sites" title="Link to this definition">¶</a></dt>
<dd><p>Compute the mean density of the target map at coordinates for non-hydrogen
atoms.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.box_build_refine_base.only_residue">
<span class="sig-name descname"><span class="pre">only_residue</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.only_residue" title="Link to this definition">¶</a></dt>
<dd><p>Returns the atom_group object corresponding to the original selection.
Will raise an exception if the selection covers more than one atom_group.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.box_build_refine_base.real_space_refine">
<span class="sig-name descname"><span class="pre">real_space_refine</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">selection</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.real_space_refine" title="Link to this definition">¶</a></dt>
<dd><p>Run simple real-space refinement, returning the optimized weight for the
map target (wx).</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.box_build_refine_base.restrain_atoms">
<span class="sig-name descname"><span class="pre">restrain_atoms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">selection</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_sigma</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">limit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reset_first</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">top_out_potential</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.restrain_atoms" title="Link to this definition">¶</a></dt>
<dd><p>Apply harmonic reference restraints to the selected atoms, wiping out
any previously existing harmonic restraints.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.box_build_refine_base.update_coordinates">
<span class="sig-name descname"><span class="pre">update_coordinates</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sites_cart</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.update_coordinates" title="Link to this definition">¶</a></dt>
<dd><p>Set coordinates for the boxed xray_structure and pdb_hierarchy objects.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.box_build_refine_base.update_original_coordinates">
<span class="sig-name descname"><span class="pre">update_original_coordinates</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.update_original_coordinates" title="Link to this definition">¶</a></dt>
<dd><p>Copies the new coordinates from the box to the selected atoms in the
original xray_structure object, and returns the (complete) list of
coordinates.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.box_build_refine_base.update_sites_from_pdb_atoms">
<span class="sig-name descname"><span class="pre">update_sites_from_pdb_atoms</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.box_build_refine_base.update_sites_from_pdb_atoms" title="Link to this definition">¶</a></dt>
<dd><p>Update the xray_structure coordinates after manipulating the PDB hierarchy.</p>
</dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.extract_iselection">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">extract_iselection</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_objects</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.extract_iselection" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.generate_sidechain_clusters">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">generate_sidechain_clusters</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">residue</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mon_lib_srv</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.generate_sidechain_clusters" title="Link to this definition">¶</a></dt>
<dd><p>Extract Chi angle indices (including rotation axis) from the atom_group</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.get_difference_maps">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">get_difference_maps</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fmodel</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">resolution_factor</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.25</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.get_difference_maps" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.get_model_map_stats">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">get_model_map_stats</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">selection</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_map</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_map</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">unit_cell</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sites_cart</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_atoms</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">local_sampling</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.get_model_map_stats" title="Link to this definition">¶</a></dt>
<dd><p>Collect basic statistics for a model map and some target map (usually an
mFo-DFc map), including CC, mean, and minimum density at the atomic
positions.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.get_nearby_water_selection">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">get_nearby_water_selection</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_hierarchy</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xray_structure</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">selection</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">selection_buffer_radius</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.get_nearby_water_selection" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.get_non_hydrogen_atom_indices">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">get_non_hydrogen_atom_indices</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_object</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.get_non_hydrogen_atom_indices" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.get_window_around_residue">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">get_window_around_residue</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">residue</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.get_window_around_residue" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.is_stub_residue">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">is_stub_residue</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atom_group</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.is_stub_residue" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.iter_residue_groups">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">iter_residue_groups</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">hierarchy</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.iter_residue_groups" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="mmtbx.building.local_density_quality">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">local_density_quality</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fofc_map</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">two_fofc_map</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sites_cart</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atom_names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">unit_cell</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atom_selection</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xray_structure</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radius</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2.5</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.11)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.local_density_quality.atoms_below_fofc_map_level">
<span class="sig-name descname"><span class="pre">atoms_below_fofc_map_level</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sigma_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-2.5</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.atoms_below_fofc_map_level" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.local_density_quality.atoms_below_two_fofc_map_level">
<span class="sig-name descname"><span class="pre">atoms_below_two_fofc_map_level</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sigma_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-2.5</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.atoms_below_two_fofc_map_level" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.local_density_quality.atoms_outside_density">
<span class="sig-name descname"><span class="pre">atoms_outside_density</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwds</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.atoms_outside_density" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.local_density_quality.density_at_atom">
<span class="sig-name descname"><span class="pre">density_at_atom</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atom_name</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.density_at_atom" title="Link to this definition">¶</a></dt>
<dd><p>Return a simple object containing density levels for the named atom,
or None if no match was found.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.local_density_quality.fraction_of_nearby_grid_points_above_cutoff">
<span class="sig-name descname"><span class="pre">fraction_of_nearby_grid_points_above_cutoff</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sigma_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2.5</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.fraction_of_nearby_grid_points_above_cutoff" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.local_density_quality.max_fofc_value">
<span class="sig-name descname"><span class="pre">max_fofc_value</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.max_fofc_value" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.local_density_quality.number_of_atoms_below_fofc_map_level">
<span class="sig-name descname"><span class="pre">number_of_atoms_below_fofc_map_level</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">*</span></span><span class="n"><span class="pre">args</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwds</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.number_of_atoms_below_fofc_map_level" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.local_density_quality.number_of_atoms_below_two_fofc_map_level">
<span class="sig-name descname"><span class="pre">number_of_atoms_below_two_fofc_map_level</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">*</span></span><span class="n"><span class="pre">args</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwds</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.number_of_atoms_below_two_fofc_map_level" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.local_density_quality.number_of_atoms_outside_density">
<span class="sig-name descname"><span class="pre">number_of_atoms_outside_density</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">*</span></span><span class="n"><span class="pre">args</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwds</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.number_of_atoms_outside_density" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="mmtbx.building.local_density_quality.show_atoms_outside_density">
<span class="sig-name descname"><span class="pre">show_atoms_outside_density</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwds</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.local_density_quality.show_atoms_outside_density" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.remove_sidechain_atoms">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">remove_sidechain_atoms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_objects</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.remove_sidechain_atoms" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.reprocess_pdb">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">reprocess_pdb</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_hierarchy</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">crystal_symmetry</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cif_objects</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.reprocess_pdb" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.residue_density_quality">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">residue_density_quality</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atom_group</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">unit_cell</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">two_fofc_map</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fofc_map</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.residue_density_quality" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.residues_are_adjacent">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">residues_are_adjacent</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">residue1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">residue2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_sep</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2.5</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.residues_are_adjacent" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.run_real_space_annealing">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">run_real_space_annealing</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">xray_structure</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_hierarchy</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">selection</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_map</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">d_min</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">processed_pdb_file</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cif_objects</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">()</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">resolution_factor</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.25</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">params</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">wc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_map_rsr</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rsr_after_anneal</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_sigma</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">debug</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.run_real_space_annealing" title="Link to this definition">¶</a></dt>
<dd><p>Run simulated annealing with a real-space target.  For maximum flexibility,
a separate map may be used for the initial real-space refinement step.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="mmtbx.building.show_chain_resseq_ranges">
<span class="sig-prename descclassname"><span class="pre">mmtbx.building.</span></span><span class="sig-name descname"><span class="pre">show_chain_resseq_ranges</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">residues</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out=&lt;colorama.ansitowin32.StreamWrapper</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">prefix=''</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#mmtbx.building.show_chain_resseq_ranges" title="Link to this definition">¶</a></dt>
<dd><p>Given a list of residues (either residue_group or atom_group objects),
print a summary of the chains and residue ranges present.</p>
</dd></dl>

</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <div>
    <h3><a href="../index.html">Table of Contents</a></h3>
    <ul>
<li><a class="reference internal" href="#">mmtbx.building package</a><ul>
<li><a class="reference internal" href="#submodules">Submodules</a></li>
<li><a class="reference internal" href="#module-mmtbx.building.extend_sidechains">mmtbx.building.extend_sidechains module</a><ul>
<li><a class="reference internal" href="#mmtbx.building.extend_sidechains.conformation_scorer"><code class="docutils literal notranslate"><span class="pre">conformation_scorer</span></code></a><ul>
<li><a class="reference internal" href="#mmtbx.building.extend_sidechains.conformation_scorer.apply_final"><code class="docutils literal notranslate"><span class="pre">conformation_scorer.apply_final()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.extend_sidechains.conformation_scorer.reset"><code class="docutils literal notranslate"><span class="pre">conformation_scorer.reset()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.extend_sidechains.conformation_scorer.update"><code class="docutils literal notranslate"><span class="pre">conformation_scorer.update()</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#mmtbx.building.extend_sidechains.correct_sequence"><code class="docutils literal notranslate"><span class="pre">correct_sequence()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.extend_sidechains.extend_and_refine"><code class="docutils literal notranslate"><span class="pre">extend_and_refine</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.extend_sidechains.extend_protein_model"><code class="docutils literal notranslate"><span class="pre">extend_protein_model()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.extend_sidechains.extend_residue"><code class="docutils literal notranslate"><span class="pre">extend_residue()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.extend_sidechains.prefilter"><code class="docutils literal notranslate"><span class="pre">prefilter</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.extend_sidechains.refit_residues"><code class="docutils literal notranslate"><span class="pre">refit_residues()</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#module-mmtbx.building.make_library">mmtbx.building.make_library module</a><ul>
<li><a class="reference internal" href="#mmtbx.building.make_library.ensembler"><code class="docutils literal notranslate"><span class="pre">ensembler</span></code></a><ul>
<li><a class="reference internal" href="#mmtbx.building.make_library.ensembler.align_model"><code class="docutils literal notranslate"><span class="pre">ensembler.align_model()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.ensembler.as_multi_model_hierarchy"><code class="docutils literal notranslate"><span class="pre">ensembler.as_multi_model_hierarchy()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.ensembler.realign"><code class="docutils literal notranslate"><span class="pre">ensembler.realign()</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#mmtbx.building.make_library.extract_and_superpose"><code class="docutils literal notranslate"><span class="pre">extract_and_superpose()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.extract_peptide_fragments_by_sequence"><code class="docutils literal notranslate"><span class="pre">extract_peptide_fragments_by_sequence()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.extract_related_models"><code class="docutils literal notranslate"><span class="pre">extract_related_models</span></code></a><ul>
<li><a class="reference internal" href="#mmtbx.building.make_library.extract_related_models.examine_model"><code class="docutils literal notranslate"><span class="pre">extract_related_models.examine_model()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.extract_related_models.results"><code class="docutils literal notranslate"><span class="pre">extract_related_models.results()</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#mmtbx.building.make_library.fetch_similar_pdb_ids"><code class="docutils literal notranslate"><span class="pre">fetch_similar_pdb_ids()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.find_similar_chains"><code class="docutils literal notranslate"><span class="pre">find_similar_chains()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.join_hierarchies"><code class="docutils literal notranslate"><span class="pre">join_hierarchies()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.load_all_models_in_directory"><code class="docutils literal notranslate"><span class="pre">load_all_models_in_directory()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.load_pdb_models"><code class="docutils literal notranslate"><span class="pre">load_pdb_models()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.load_related_models_from_pdb"><code class="docutils literal notranslate"><span class="pre">load_related_models_from_pdb()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.load_related_models_in_directory"><code class="docutils literal notranslate"><span class="pre">load_related_models_in_directory()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.related_chain"><code class="docutils literal notranslate"><span class="pre">related_chain</span></code></a><ul>
<li><a class="reference internal" href="#mmtbx.building.make_library.related_chain.chain_id"><code class="docutils literal notranslate"><span class="pre">related_chain.chain_id</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.related_chain.identity"><code class="docutils literal notranslate"><span class="pre">related_chain.identity</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.related_chain.pdb_hierarchy"><code class="docutils literal notranslate"><span class="pre">related_chain.pdb_hierarchy</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.make_library.related_chain.source_info"><code class="docutils literal notranslate"><span class="pre">related_chain.source_info</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#mmtbx.building.make_library.selection_by_symmetry"><code class="docutils literal notranslate"><span class="pre">selection_by_symmetry()</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#module-mmtbx.building">Module contents</a><ul>
<li><a class="reference internal" href="#mmtbx.building.atom_group_as_hierarchy"><code class="docutils literal notranslate"><span class="pre">atom_group_as_hierarchy()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.box_build_refine_base"><code class="docutils literal notranslate"><span class="pre">box_build_refine_base</span></code></a><ul>
<li><a class="reference internal" href="#mmtbx.building.box_build_refine_base.anneal"><code class="docutils literal notranslate"><span class="pre">box_build_refine_base.anneal()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.box_build_refine_base.cc_model_map"><code class="docutils literal notranslate"><span class="pre">box_build_refine_base.cc_model_map()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.box_build_refine_base.fit_residue_in_box"><code class="docutils literal notranslate"><span class="pre">box_build_refine_base.fit_residue_in_box()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.box_build_refine_base.geometry_minimization"><code class="docutils literal notranslate"><span class="pre">box_build_refine_base.geometry_minimization()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.box_build_refine_base.get_selected_sites"><code class="docutils literal notranslate"><span class="pre">box_build_refine_base.get_selected_sites()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.box_build_refine_base.mean_density_at_sites"><code class="docutils literal notranslate"><span class="pre">box_build_refine_base.mean_density_at_sites()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.box_build_refine_base.only_residue"><code class="docutils literal notranslate"><span class="pre">box_build_refine_base.only_residue()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.box_build_refine_base.real_space_refine"><code class="docutils literal notranslate"><span class="pre">box_build_refine_base.real_space_refine()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.box_build_refine_base.restrain_atoms"><code class="docutils literal notranslate"><span class="pre">box_build_refine_base.restrain_atoms()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.box_build_refine_base.update_coordinates"><code class="docutils literal notranslate"><span class="pre">box_build_refine_base.update_coordinates()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.box_build_refine_base.update_original_coordinates"><code class="docutils literal notranslate"><span class="pre">box_build_refine_base.update_original_coordinates()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.box_build_refine_base.update_sites_from_pdb_atoms"><code class="docutils literal notranslate"><span class="pre">box_build_refine_base.update_sites_from_pdb_atoms()</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#mmtbx.building.extract_iselection"><code class="docutils literal notranslate"><span class="pre">extract_iselection()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.generate_sidechain_clusters"><code class="docutils literal notranslate"><span class="pre">generate_sidechain_clusters()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.get_difference_maps"><code class="docutils literal notranslate"><span class="pre">get_difference_maps()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.get_model_map_stats"><code class="docutils literal notranslate"><span class="pre">get_model_map_stats()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.get_nearby_water_selection"><code class="docutils literal notranslate"><span class="pre">get_nearby_water_selection()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.get_non_hydrogen_atom_indices"><code class="docutils literal notranslate"><span class="pre">get_non_hydrogen_atom_indices()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.get_window_around_residue"><code class="docutils literal notranslate"><span class="pre">get_window_around_residue()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.is_stub_residue"><code class="docutils literal notranslate"><span class="pre">is_stub_residue()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.iter_residue_groups"><code class="docutils literal notranslate"><span class="pre">iter_residue_groups()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.local_density_quality"><code class="docutils literal notranslate"><span class="pre">local_density_quality</span></code></a><ul>
<li><a class="reference internal" href="#mmtbx.building.local_density_quality.atoms_below_fofc_map_level"><code class="docutils literal notranslate"><span class="pre">local_density_quality.atoms_below_fofc_map_level()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.local_density_quality.atoms_below_two_fofc_map_level"><code class="docutils literal notranslate"><span class="pre">local_density_quality.atoms_below_two_fofc_map_level()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.local_density_quality.atoms_outside_density"><code class="docutils literal notranslate"><span class="pre">local_density_quality.atoms_outside_density()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.local_density_quality.density_at_atom"><code class="docutils literal notranslate"><span class="pre">local_density_quality.density_at_atom()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.local_density_quality.fraction_of_nearby_grid_points_above_cutoff"><code class="docutils literal notranslate"><span class="pre">local_density_quality.fraction_of_nearby_grid_points_above_cutoff()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.local_density_quality.max_fofc_value"><code class="docutils literal notranslate"><span class="pre">local_density_quality.max_fofc_value()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.local_density_quality.number_of_atoms_below_fofc_map_level"><code class="docutils literal notranslate"><span class="pre">local_density_quality.number_of_atoms_below_fofc_map_level()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.local_density_quality.number_of_atoms_below_two_fofc_map_level"><code class="docutils literal notranslate"><span class="pre">local_density_quality.number_of_atoms_below_two_fofc_map_level()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.local_density_quality.number_of_atoms_outside_density"><code class="docutils literal notranslate"><span class="pre">local_density_quality.number_of_atoms_outside_density()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.local_density_quality.show_atoms_outside_density"><code class="docutils literal notranslate"><span class="pre">local_density_quality.show_atoms_outside_density()</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#mmtbx.building.remove_sidechain_atoms"><code class="docutils literal notranslate"><span class="pre">remove_sidechain_atoms()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.reprocess_pdb"><code class="docutils literal notranslate"><span class="pre">reprocess_pdb()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.residue_density_quality"><code class="docutils literal notranslate"><span class="pre">residue_density_quality()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.residues_are_adjacent"><code class="docutils literal notranslate"><span class="pre">residues_are_adjacent()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.run_real_space_annealing"><code class="docutils literal notranslate"><span class="pre">run_real_space_annealing()</span></code></a></li>
<li><a class="reference internal" href="#mmtbx.building.show_chain_resseq_ranges"><code class="docutils literal notranslate"><span class="pre">show_chain_resseq_ranges()</span></code></a></li>
</ul>
</li>
</ul>
</li>
</ul>

  </div>
  <div>
    <h4>Previous topic</h4>
    <p class="topless"><a href="mmtbx.html"
                          title="previous chapter">mmtbx - macromolecular crystallography</a></p>
  </div>
  <div>
    <h4>Next topic</h4>
    <p class="topless"><a href="mmtbx.bulk_solvent.html"
                          title="next chapter">mmtbx.bulk_solvent package</a></p>
  </div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/mmtbx/mmtbx.building.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="../py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="mmtbx.bulk_solvent.html" title="mmtbx.bulk_solvent package"
             >next</a> |</li>
        <li class="right" >
          <a href="mmtbx.html" title="mmtbx - macromolecular crystallography"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">CCTBX Developer documentation</a> &#187;</li>
          <li class="nav-item nav-item-1"><a href="mmtbx.html" >mmtbx - macromolecular crystallography</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">mmtbx.building package</a></li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
    &#169; Copyright 2014, University of California.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 7.2.5.
    </div>
  </body>
</html>